dali/core/mm/default_resources.cc

```diff
@@ -235,6 +235,9 @@ inline std::shared_ptr<device_async_resource> CreateDefaultDeviceResource() {
   CUDA_CALL(cudaGetDevice(&device_id));
   if (MMEnv::get().use_cuda_malloc_async) {
     #if CUDA_VERSION >= 11020
+      if (!cuda_malloc_async_memory_resource::is_supported(device_id))
```

Comment:
So if user wants async but it is not supported we want to verbosely fail, right? No silent fallbacks or something.

Reply:
Yes - it's not enabled by default, so I think we can fail if the user specifically requests this method.